### PR TITLE
Single signings scripts improvements

### PR DESCRIPTION
**Pull request by lsubash**
Udated error handling for SBL Key dir and error
messages to guide user to use GenerateKeys tool.

Signed-off-by: Subash Lakkimsetti <subash.lakkimsetti@intel.com>

### CODE REVIEW

File: BootloaderCorePkg/Tools/BuildUtility.py [hunk 1]
```diff
@@ -862,7 +862,11 @@ def check_for_git():
 
 def check_for_slimbootkeydir():
     if not os.path.exists(os.environ.get('SBL_KEY_DIR')):
-        raise Exception ("SBL_KEY_DIR is not valid. Set correct SBL_KEY_DIR path !!")
+        print ("!!! PRE_REQUISTE: PATH TO SBL_KEY_DIR HAS TO BE SET WITH SBL KEY DIRECTORY !!! \n"
```
Comment (sagraw2) [inline BootloaderCorePkg/Tools/BuildUtility.py:865]: Fix the various typos: PRE-REQUISITE, AVAILABLE etc

Comment (sagraw2) [reply BootloaderCorePkg/Tools/BuildUtility.py:865]: GenerateKeys script does not set the environment variable. 
Do we need to be pedantic here in our error message and provide example on how to set env var on Windows and Linux systesm?

File: BootloaderCorePkg/Tools/SingleSign.py [hunk 2]
```diff
@@ -110,45 +111,58 @@ def get_key_id (priv_key):
...
     if priv_key is not None:
         if (priv_key in SIGNING_KEY):
             # Generate key file name from key id
             priv_key_file = SIGNING_KEY[priv_key] + '_' + KEY_SIZE_TYPE +'.pem'
         else:
-            raise Exception('KEY_ID %s is not found!' % priv_key)
+            raise Exception('KEY_ID %s is not found in supported KEY IDs!!' % priv_key)
     elif check_file_pem_format(in_key) == True:
-        # check if file is of pem format
+        # check if file name is privided in pem format
```
Comment (sagraw2) [inline BootloaderCorePkg/Tools/SingleSign.py:145]: provided

File: BootloaderCorePkg/Tools/BuildUtility.py (hunk 1 above)
Comment (mauricema) [inline BootloaderCorePkg/Tools/BuildUtility.py:865]: Let us separate 2 cases
- SBL_KEY_DIR is not set.  So prompt user the default PATH is used.
- SBL_KEY_DIR is a valid path, but there is no keys in it.   Prompt the msg below.

File: BootloaderCorePkg/Tools/BuildUtility.py [hunk 3]
```diff
@@ -862,7 +862,11 @@ def check_for_git():
 
 def check_for_slimbootkeydir():
     if not os.path.exists(os.environ.get('SBL_KEY_DIR')):
-        raise Exception ("SBL_KEY_DIR is not valid. Set correct SBL_KEY_DIR path !!")
+        print ("!!! PRE_REQUISTE: PATH TO SBL_KEY_DIR HAS TO BE SET WITH SBL KEY DIRECTORY !!! \n"
+                "!!! ERROR: SBL_KEY_DIR IS SET TO %s DO NOT EXIST!!! \n"
```
Comment (mauricema) [inline BootloaderCorePkg/Tools/BuildUtility.py:866]: SBL_KEY_DIR IS SET TO %s, but required keys were not found !!!

File: BootloaderCorePkg/Tools/SingleSign.py [hunk 4]
```diff
@@ -49,9 +49,10 @@
 
 
 def print_message_sbl_key_dir ():
```
Comment (mauricema) [inline BootloaderCorePkg/Tools/SingleSign.py:51]: print_sbl_key_dir

File: BootloaderCorePkg/Tools/SingleSign.py [hunk 5]
```diff
@@ -49,9 +49,10 @@
 
 
 def print_message_sbl_key_dir ():
-    print ("Pre-requiste: SBL_KEY_DIR environment variable has to be set!")
-    print ("SBL_KEY_DIR is path to keys used for the project!")
-    print ("For Keys generation follow GenerateKeys.py available in tool directory!")
+    print ("Pre-requiste: SBL_KEY_DIR environment variable has to be set!!")
+    print ("SBL_KEY_DIR is path to keys used for the project!!")
+    print ("TO GENERATE KEYS USE GenerateKeys.py AVAIABLE IN BootloaderCorePkg/Tools DIRECTORY !!")
+    print ("Run python BootloaderCorePkg/Tools/GenerateKeys.py -k $SBL_KEY_DIR!!")
```
Comment (mauricema) [inline BootloaderCorePkg/Tools/SingleSign.py:55]: we have duplicated info for key generate ?  Can we just have one place for this msg ?

File: BootloaderCorePkg/Tools/SingleSign.py [hunk 6]
```diff
@@ -110,45 +111,58 @@ def get_key_id (priv_key):
...
         priv_key = os.path.join (sbl_key_dir, priv_key_file)
     except:
         raise Exception('priv_key is not found %s!' % priv_key)
 
+    # Check for priv_key construted based on KET ID exists in specified path
+    if not os.path.isfile(priv_key):
+        message =  ("!!! ERROR: Key file corresponding to '%s' do not exist in key store at '%s'!! !!! \n"
+                    "!!! Check SBL_KEY_DIR setting!!! \n"
+                    "!!! TO GENERATE KEYS USE GenerateKeys.py AVAIABLE IN BootloaderCorePkg/Tools DIRECTORY !!! \n"
+                    "!!! Run python BootloaderCorePkg/Tools/GenerateKeys.py -k %s !!! " % (in_key, sbl_key_dir, sbl_key_dir))
```
Comment (mauricema) [inline BootloaderCorePkg/Tools/SingleSign.py:163]: try to do this in a single place.

File: BootloaderCorePkg/Tools/BuildUtility.py (hunk 1 above)
Comment (lsubash) [reply BootloaderCorePkg/Tools/BuildUtility.py]: Fixed typos and updated messages to set env var

Comment (lsubash) [reply BootloaderCorePkg/Tools/BuildUtility.py]: Addressed both cases.

**Comment by lsubash** [comment]
Updated PR,
- Added "MESSAGE_SBL_KEY_DIR" for pre-requisite messages and reusing the same 
- Fixed typos

**Pull request merged**